### PR TITLE
Use Market Data in TokenDetails

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

File: packages/app/components/TopNav.tsx [hunk 1]
```diff
@@ -71,7 +72,6 @@ export function TopNav({
     // always pop to the base path. e.g. /account/settings/edit-profile -> /account
     if (hasSubrouteParam) {
       setTokenParam(undefined)
-      return
```
Comment (0xBigBoss) [inline packages/app/components/TopNav.tsx:74]: honestly this top nav is turning into a dumpster fire. it's about time we write some unit tests.

File: packages/app/components/TopNav.tsx [hunk 2]
```diff
@@ -83,10 +83,6 @@ export function TopNav({
   const isSubRoute = !noSubroute && parts.length > 1
 
   const selectedCoin = coins.find((c) => c.token === tokenParam)
```
Comment (0xBigBoss) [inline packages/app/components/TopNav.tsx:85]: let's move this into a hook.

File: packages/app/components/TopNav.tsx [hunk 3]
```diff
@@ -170,14 +171,52 @@ export function TopNav({
...
+              display={selectedCoin && !media.gtLg ? 'none' : 'flex'}
               als={'center'}
             >
               {header}
             </H2>
           )}
           <XStack>{renderButton()}</XStack>
         </Container>
-        <Separator w={'100%'} borderColor="$decay" $gtLg={{ display: 'none' }} />
+        {tokenParam && !media.gtLg && selectedCoin && selectedCoin.label !== 'USDC' ? (
```
Comment (0xBigBoss) [inline packages/app/components/TopNav.tsx:182]: wrap this logic in another component please.

**Review by 0xBigBoss — APPROVED** [review]
lgtm but our lack of testing on top nav worries me

**Comment by 0xBigBoss** [comment]
also a little bit of weirdness on this size screen. should the market data float all the way to the right?
Screenshot 2024-04-18 at 09 56 37

File: packages/app/components/TopNav.tsx (hunk 1 above)
Comment (youngkidwarrior) [reply packages/app/components/TopNav.tsx:74]: Yes, I'm thinking the same thing. Haven't had a chance to bring it up. I'm going to spend time reworking the whole thing.

One main difficulty is that we pass TopNav in `getLayout`, which isn't a React component. So we can't use any logic around it when it is renders. I have some good ideas to improve it

File: packages/app/components/TopNav.tsx (hunk 3 above)
Comment (youngkidwarrior) [reply packages/app/components/TopNav.tsx:182]: Let's push this for now so we can get updates out by Friday. It's smelly as fuck, but it works

I'm going to do a massive refactor of the TopNav logic and scaffolding. I'll fix all this when I make those changes.

**Comment by youngkidwarrior** [comment]
> also a little bit of weirdness on this size screen. should the market data float all the way to the right?

Yeah, this was a weird screen. when dealing with the `Container`. Let me try getting a fix out

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Apr 22, 5:21 PM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/329).

**Pull request merged**